### ISSUE TITLE
[1.20 regression] "Statement is unreachable" for `dict` and `ResponseHeaders` compare

### ISSUE THREAD

**Issue by sobolevn** [python/mypy#21265]
A lot of my `assert` statements in tests are reported as unreachable even with `1.20.1`,
because of this check:

```python
assert response.headers == {
    'Cache-Control': 'no-cache',
    'Content-Type': 'text/event-stream',
    'X-Accel-Buffering': 'no',
    'Connection': 'keep-alive',
}
```

Where `response.headers` is `ResponseHeaders` https://github.com/typeddjango/django-stubs/blob/1ac68b440b14cc3ac913462addb6fd8c46b20460/django-stubs/http/response.pyi#L16

Tests pass in runtime, however I have 178 errors in 53 files :(

<img width="788" height="533" alt="Image" src="https://github.com/user-attachments/assets/403e2d9d-aa55-43d6-8e3e-f55697270a31" />

Link to the test: https://github.com/wemake-services/django-modern-rest/blob/5cbea86f32cd3a00095949295e5004d2d7bd66c8/tests/test_unit/test_streaming/test_sse/test_class_sse.py#L113

CC @hauntsaninja

**Comment by hauntsaninja** [comment]
AFK but wasn't this fixed by the update to django-stubs https://github.com/python/mypy/issues/21132#issuecomment-4172289163

**Comment by sobolevn** [comment]
Exactly 🤦 
I forgot to release this feature. Sorry :)
